crates/types/src/nodes_config.rs

```diff
@@ -156,6 +156,15 @@ impl NodesConfiguration {
         self.version = version;
     }
 
+    /// Removes the node with the given `id` by inserting a tombstone for it.
+    pub fn remove_node(&mut self, id: impl Into<NodeId>) {
+        let node_id = id.into();
+        // only keep tombstones for known nodes
+        self.nodes
+            .entry(node_id.id())
+            .and_modify(|value| *value = MaybeNode::Tombstone);
+    }
```

Comment:
Can we call this `remove_node_unchecked` and leave a comment about when this can be used?